client/keys/add.go

```diff
@@ -75,6 +75,23 @@ the flag --nosort is set.
 	return cmd
 }
 
+func getKeybase(dryrun bool) (keys.Keybase, error) {
+	if dryrun {
+		return keys.NewInMemory(), nil
+	}
+
+	return NewKeyBaseFromHomeFlag()
+}
+
+func runAddCmd(cmd *cobra.Command, args []string) error {
+	kb, err := getKeybase(viper.GetBool(flagDryRun))
+	if err != nil {
+		return err
+	}
+
+	return RunAddCmd(cmd, args, kb)
```

Comment:
Yeah, I agree, the rationale was that I did not want to require an application to duplicate all of the command logic just to add options to the keybase. Ideally I would have liked to restructure the flow of this command, to split the logic into generating the key bytes from the command parameters and saving the key, so that instead of options passed to the keybase, the key type that is saved could just be defined in the command. I didn't want to make breaking changes though, so I will try to think of a better way of doing this without making breaking changes

Reply:
I see and appreciate the rationale, though I'm not convinced about the implementation. I like see a suboptimal design flow better than a potentially broken one.

Reply:
Oh I'm sorry, I missed this comment. I'll explain what I meant by a different flow by some vague pseudocode and hopefully this will be sufficient to get the point across:

currently the code vaguely does:
```
addcommand() {
  info ... = kb.CreateAccount(...)
}
...) CreateAccount(..) {
 ...
  derivedBz = ComputeDerivedKey(seed,...)
  writeKey(...derivedBz)
}
```

Where it would be more ideal to pass the bytes back to the addcommand to be able to derive whatever implementation there instead of having options. It doesn't make much sense to me why the key is derived and written all in one flow anyway. This would have removed the need for this PR altogether because I could have just overriden the add command and been done with it. Example in pseudo:
```
addcommand() {
  derivedBz = ComputeDerivedKey(...)
  writeKey(...derivedBz)
}
```